manager/orchestrator/task.go

```diff
@@ -93,6 +93,32 @@ func IsTaskDirty(s *api.Service, t *api.Task) bool {
 		(t.Endpoint != nil && !reflect.DeepEqual(s.Spec.Endpoint, t.Endpoint.Spec))
 }
 
+// IsTaskPlacementConstraintsOnlyDirty checks if the Placement field alone
+// in the spec has changed.
+func IsTaskPlacementConstraintsOnlyDirty(s *api.Service, t *api.Task) bool {
+	if s.Spec.Task.Placement == nil && t.Spec.Placement == nil {
+		return false
+	}
+
+	// Make a deep copy of the service and task spec for the comparison.
+	serviceTaskSpec := *s.Spec.Task.Copy()
```

Comment:
@anshulpundir does this copy over any map objects in the spec correctly?

Reply:
Yes